sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/StructFiltersSuite.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.sources
+import org.apache.spark.sql.sources.{AlwaysFalse, AlwaysTrue, Filter}
+import org.apache.spark.sql.types.{IntegerType, StructType}
+import org.apache.spark.unsafe.types.UTF8String
+
+abstract class StructFiltersSuite extends SparkFunSuite {
+
+  def createFilters(filters: Seq[sources.Filter], schema: StructType): StructFilters
```

Comment:
To be more explicit, `def` -> `protected def`?

Reply:
If you extend an abstract class with methods without implementation, the compilation fails in any case, right?

Reply:
I will add `protected` but it is unnecessary noise, from my point of view.

Reply:
@MaxGekk . There is a reason why Java/Scala have `protected`. You may want to use `public` always, but it's not a good practice in terms of code-wise or review-wise. You had better change your point of view to become a committer.

Reply:
You mixes 2 things - scope and what should be implemented in child classes. `protected` doesn't indicate that a method must be implemented in a child class because it can have an implementation in the parent class. 

> You had better change your point of view to become a committer.

Thank you, now I know what blocks me ;-)